crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -723,6 +723,13 @@ impl<'a> Visitor<'a> for Checker<'a> {
                 // Visit the decorators and arguments, but avoid the body, which will be
                 // deferred.
                 for decorator in decorator_list {
+                    if self
+                        .semantic
+                        .match_typing_expr(&decorator.expression, "no_type_check")
+                    {
+                        self.semantic.flags |= SemanticModelFlags::NO_TYPE_CHECK;
+                    }
+
                     self.visit_decorator(decorator);
```

Comment:
I'd flip the order here so that Ruff's and Red Knot's behavior is identical. 


```suggestion
                    self.visit_decorator(decorator);
                    if self
                        .semantic
                        .match_typing_expr(&decorator.expression, "no_type_check")
                    {
                        self.semantic.flags |= SemanticModelFlags::NO_TYPE_CHECK;
                    }
```